devito/operator.py

```diff
@@ -124,30 +123,33 @@ def __init__(self, expressions, **kwargs):
         # Finish instantiation
         super(Operator, self).__init__(self.name, nodes, 'int', parameters, ())
 
-    @cached_property
-    def _argument_defaults(self):
+    def _argument_defaults(self, arguments):
         """
         Derive all default values from parameters and ensure uniqueness.
         """
         default_args = ArgumentMap()
         for p in self.input:
-            default_args.update(p.argument_defaults())
+            if p.name not in arguments:
+                default_args.update(p.argument_defaults())
+            else:
+                default_args.update(p.argument_defaults(data=False))
```

Comment:
I don't like this at all, though I think I see why you had to do it -- to hit this line:
https://github.com/opesci/devito/pull/486/files#diff-392d49fabd2c76220fb6a2042dc1f460R537